### PR TITLE
Blocked - don't merge - DIV-3771 - initial Helm chart files

### PR DESCRIPTION
**Pull request by hosainnet**
# Description

Use Helm Charts for deployment. This is similar to our AKS deployment. The difference is that we will be able list the dependencies that our service requires so that in the future we can build/deploy a complete stack easily, which will also help for the new HMCTS Demo env

Fixes #DIV-3771

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

App deployed successfully. Checked the pod on the AKS dashboard

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by qzhou-hmcts** [comment]
Should we apply this on master?

**Comment by hosainnet** [comment]
@qzhou-hmcts similar to previously with AKS pipeline, this will not take effect on master, just PRs

File: charts/div-cms/values.aat.template.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+java:
```
Comment (KevinJCross) [inline charts/div-cms/values.aat.template.yaml]: You should move all the stuff that is not aat specific into the values.template.yaml
These would include:

AUTH_PROVIDER_SERVICE_CLIENT_KEY
AUTH2_CLIENT_SECRET
DRAFT_STORE_API_ENCRYPTION_KEY_VALUE
APPLICATION_INSIGHTS_IKEY
....

basically almost everything that is not a aat url.

File: charts/div-cms/values.template.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+java:
+    applicationPort: 4010
+    # Don't modify below here
+    image: ${IMAGE_NAME}
+    ingressHost: ${SERVICE_FQDN}
+    environment:
+        AUTH_PROVIDER_SERVICE_CLIENT_KEY: ${AUTH_PROVIDER_SERVICE_CLIENT_KEY}
+        AUTH2_CLIENT_SECRET: ${AUTH2_CLIENT_SECRET}
+        DRAFT_STORE_API_ENCRYPTION_KEY_VALUE: ${DRAFT_STORE_API_ENCRYPTION_KEY_VALUE}
+        IDAM_CASEWORKER_PASSWORD: ${IDAM_CASEWORKER_PASSWORD}
```
Comment (luigibk) [inline charts/div-cms/values.template.yaml]: These 4 values above will not be replaced by the pipeline (those are the only substitutions in this file). It looks like they are vault secrets, so they should be coming from Jenkinsfile.

Comment (hosainnet) [reply charts/div-cms/values.template.yaml]: They're already extracted in `Jenkinsfile_CNP`, does that mean they're redundent here?

File: charts/div-cms/values.aat.template.yaml (hunk 1 above)
Comment (hosainnet) [reply charts/div-cms/values.aat.template.yaml]: Makes sense 👍

File: charts/div-cms/values.template.yaml (hunk 2 above)
Comment (luigibk) [reply charts/div-cms/values.template.yaml]: No, if they are added to the environment by Jenkinsfile_CNP adds, then that should work and you can ignore my comment.

**Comment by hosainnet** [comment]
Closing for now as it's not neede at the moment

**Pull request closed without merging**